ci-operator/step-registry/hypershift/conformance/hypershift-conformance-chain.yaml

```diff
@@ -15,16 +15,23 @@ chain:
       trap cleanup EXIT
 
       function patch_image_streams() {
+        # with OCP <= 4.17,
         # imagestreams on the hosted cluster are ignoring
         # the CA injected at creation/(installation) time.
         # see: https://issues.redhat.com/browse/RFE-3093
         # and: https://github.com/openshift/enhancements/pull/416/commits/e46072ecf77b58c79b4bece5e0c6a5a5546a1b8a
 
-        echo "#### Add trust for the mirroring registry"
-        oc get cm -n openshift-config user-ca-bundle -o json | jq -r '.data["ca-bundle.crt"]' > /tmp/ca-bundle.crt
-        oc create configmap registry-config --from-file=${DS_REGISTRY//:/..}=/tmp/ca-bundle.crt -n openshift-config
-        oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-config"}}}' --type=merge
-        sleep 60
+        OCP_MINOR_VERSION=$(oc version | grep "Server Version" | cut -d '.' -f2)
+        if [ "$OCP_MINOR_VERSION" -ge "16" ]; then
+           oc --kubeconfig="${HYPERSHIFT_MANAGEMENT_CLUSTER_KUBECONFIG}" patch hostedclusters -n "${HYPERSHIFT_NAMESPACE}" "${CLUSTER_NAME}" --type='merge' -p '{"spec":{"configuration":{"image":{"additionalTrustedCA":{"name": "user-ca-bundle"}}}}}'
+        fi
+        if [ "$OCP_MINOR_VERSION" -le "17" ]; then
+           echo "#### Add trust for the mirroring registry"
+           oc get cm -n openshift-config user-ca-bundle -o json | jq -r '.data["ca-bundle.crt"]' > /tmp/ca-bundle.crt
+           oc create configmap registry-config --from-file=${DS_REGISTRY//:/..}=/tmp/ca-bundle.crt -n openshift-config
+           oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-config"}}}' --type=merge
+           sleep 60
+        fi
```

Comment:
I think changing it to this should meet our needs.
```
        if [ "$OCP_MINOR_VERSION" -le "17" ]; then
           echo "#### Add trust for the mirroring registry"
           oc get cm -n openshift-config user-ca-bundle -o json | jq -r '.data["ca-bundle.crt"]' > /tmp/ca-bundle.crt
           oc create configmap registry-config --from-file=${DS_REGISTRY//:/..}=/tmp/ca-bundle.crt -n openshift-config
           oc patch image.config.openshift.io/cluster --patch '{"spec":{"additionalTrustedCA":{"name":"registry-config"}}}' --type=merge
        else
          oc --kubeconfig="${HYPERSHIFT_MANAGEMENT_CLUSTER_KUBECONFIG}" patch hostedclusters -n "${HYPERSHIFT_NAMESPACE}" "${CLUSTER_NAME}" --type='merge' -p '{"spec":{"configuration":{"image":{"additionalTrustedCA":{"name": "user-ca-bundle"}}}}}'
        fi
        sleep 60
```

Reply:
done, thanks